python/src/opendp/typing.py

```diff
@@ -338,51 +346,6 @@ def parse_or_infer(
             return cls.infer(public_example)
         raise UnknownTypeException("either type_name or public_example must be passed")
 
-    @classmethod
-    def assert_is_similar(cls, expected, inferred):
-        """Assert that `inferred` is a member of the same equivalence class as `parsed`.
-
-        :param expected: the type that the data will be converted to
-        :param inferred: the type inferred from data
-        :raises TypeError: if `expected` type differs significantly from `inferred` type
-        """
-
-        ERROR_URL_298 = "https://github.com/opendp/opendp/discussions/298"
-        if isinstance(inferred, UnknownType):
-            return
-        
-        # allow extra flexibility around options, as the inferred type of an Option::<T>::Some will just be T
-        def is_option(type_name):
-            return isinstance(type_name, RuntimeType) and type_name.origin == "Option"
-        if is_option(expected):
-            expected = expected.args[0]
-            if is_option(inferred):
-                if isinstance(inferred.args[0], UnknownType):
-                    return
-                else:
-                    inferred = inferred.args[0]
```

Comment:
Did you lose this handling of options?

Reply:
Now `_convert.py` handles this in `py_to_c`:

```python
    if type_name is not None:
        # exit early with a null pointer if trying to load an Option type with a None value
        if isinstance(type_name, RuntimeType) and type_name.origin == "Option":
            if value is None:
                return
            type_name = type_name.args[0]
```

Reply:
Ah OK thanks!